spring-kafka/src/main/java/org/springframework/kafka/support/AbstractKafkaHeaderMapper.java

```diff
@@ -44,26 +47,29 @@
  */
 public abstract class AbstractKafkaHeaderMapper implements KafkaHeaderMapper {
 
-	protected final Log logger = LogFactory.getLog(getClass()); // NOSONAR
-
-	private static final List<SimplePatternBasedHeaderMatcher> NEVER_MAPPED = Arrays.asList(
-			new SimplePatternBasedHeaderMatcher("!" + KafkaHeaders.ACKNOWLEDGMENT),
-			new SimplePatternBasedHeaderMatcher("!" + KafkaHeaders.CONSUMER),
-			new SimplePatternBasedHeaderMatcher("!" + KafkaHeaders.MESSAGE_KEY),
-			new SimplePatternBasedHeaderMatcher("!" + KafkaHeaders.OFFSET),
-			new SimplePatternBasedHeaderMatcher("!" + KafkaHeaders.PARTITION_ID),
-			new SimplePatternBasedHeaderMatcher("!" + KafkaHeaders.RAW_DATA),
-			new SimplePatternBasedHeaderMatcher("!" + KafkaHeaders.RECEIVED_MESSAGE_KEY),
-			new SimplePatternBasedHeaderMatcher("!" + KafkaHeaders.RECEIVED_PARTITION_ID),
-			new SimplePatternBasedHeaderMatcher("!" + KafkaHeaders.RECEIVED_TIMESTAMP),
-			new SimplePatternBasedHeaderMatcher("!" + KafkaHeaders.RECEIVED_TOPIC),
-			new SimplePatternBasedHeaderMatcher("!" + KafkaHeaders.TIMESTAMP),
-			new SimplePatternBasedHeaderMatcher("!" + KafkaHeaders.TIMESTAMP_TYPE),
-			new SimplePatternBasedHeaderMatcher("!" + KafkaHeaders.BATCH_CONVERTED_HEADERS),
-			new SimplePatternBasedHeaderMatcher("!" + KafkaHeaders.NATIVE_HEADERS),
-			new SimplePatternBasedHeaderMatcher("!" + KafkaHeaders.TOPIC));
-
-	protected final List<SimplePatternBasedHeaderMatcher> matchers = new ArrayList<>(NEVER_MAPPED); // NOSONAR
+	protected final Log LOGGER = LogFactory.getLog(getClass()); // NOSONAR
+
+	private static final List<HeaderMatcher> NEVER_MAPPED = Collections.singletonList(
```

Comment:
Doesn't look like this constant is used any where.
Can't we just make it inline in the `matchers` bellow?